src/bun.js/bindings/ZigGlobalObject.cpp

```diff
@@ -3898,6 +3858,20 @@ uint8_t GlobalObject::drainMicrotasks()
 {
     auto& vm = this->vm();
     auto scope = DECLARE_CATCH_SCOPE(vm);
+
+    if (auto* exception = scope.exception()) [[unlikely]] {
+        if (vm.isTerminationException(exception)) [[unlikely]] {
+            return 1;
+        }
+
+#if ASSERT_ENABLED
+        scope.clearException();
+        Bun__reportError(this, JSValue::encode(exception));
```

Comment:
if we're catching it just to throw it again why are we reporting it as uncaught too?

Reply:
This is a debugging tool for us so we see exactly what exception was thrown in the logs instead of seeing that an exception was thrown. I'll add a comment.